src/sas/qtgui/Perspectives/Fitting/FittingUtilities.py

```diff
@@ -591,7 +591,10 @@ def plotPolydispersities(model, disp_models):
         # create Data1D as in residualsData1D() and fill x/y members
         # similar to FittingLogic._create1DPlot() but different data/axes
         data1d = Data1D(x = arr[0], y = arr[1])
-        data1d.xaxis('\\rm{{{}}} '.format(name.replace('_', '\_')), 'A') # correct unit?
+        xunit = ""
+        if name in model.details and len(model.details[name]):
+            xunit = model.details[name][0]
+        data1d.xaxis('\\rm{{{}}}'.format(name.replace('_', '\_')), xunit)
```

Comment:
I don't believe you need to protect against parameter name not being included in model.details.  Do you know of cases where model.details are not defined, or name is not in them?  If so, how is the polydisperse parameter showing up in the GUI?

Reply:
I find that problems are easier to debug if the program fails early. If the model interface really does change, then code which protects against unexpected inputs will remain quietly broken.

For the GUI, trap any errors that occur and display them to the user, complete with traceback so we can find out where they are.  This keeps the program robust.

The assumption that details exists and is initialized properly is no different from the unchecked assumption a few lines above in `if 0 == pars['width']` that width is a member of pars.  If we tried to check that every data structure has the correct form before performing any operation our code would become immensely unweildy.  Python has optional type checking via mypy which we can use to verify that the right types are being passed to functions.  I've added this as ticket 1180.

Minor note:  you can use the more natural `if x == 0:` in python rather than `if 0 == x:`.   Since `if x = 0:` is invalid, you don't get the accidental assignment bug that occasionally confuses C programmers.  Of course, if you have been doing defensive C coding for years then `if (0 == x)` may look more natural to you.